plugin/pkg/shoot/validator/admission.go

```diff
@@ -2040,3 +2043,65 @@ func isShootInMigrationOrRestorePhase(shoot *core.Shoot) bool {
 			shoot.Status.LastOperation.State != core.LastOperationStateSucceeded ||
 			shoot.Status.LastOperation.Type == core.LastOperationTypeMigrate)
 }
+
+func (c *validationContext) validateManagedServiceAccountIssuer(
+	a admission.Attributes,
+	secretLister kubecorev1listers.SecretLister,
+) error {
+	// Skip the validation if no managed service account issuer configuration is involved.
+	if !helper.HasManagedIssuer(c.shoot) &&
+		(c.oldShoot == nil || !helper.HasManagedIssuer(c.oldShoot)) {
+		return nil
+	}
+
+	managedIssuerConfigSecrets, err := secretLister.Secrets(v1beta1constants.GardenNamespace).List(labels.SelectorFromSet(labels.Set{
+		v1beta1constants.GardenRole: v1beta1constants.GardenRoleShootServiceAccountIssuer,
+	}))
+	if err != nil {
+		return apierrors.NewInternalError(fmt.Errorf("could not retrieve managed service account issuer config secret: %+v", err.Error()))
```

Comment:
```suggestion
		return apierrors.NewInternalError(fmt.Errorf("could not retrieve managed service account issuer config secret: %w", err.Error()))
```

Nit, `%+v` is mostly useless here. We usually use `%w` or `%s`.